### PR TITLE
chore: Migrate to npm, add rimraf devDependency

### PR DESCRIPTION
**Pull request by jnv**

### CODE REVIEW

**Comment by jnv** [comment]
_By the way_ we could use [TypeScript's project references](https://www.typescriptlang.org/docs/handbook/project-references.html) with `--clean` swithc and avoid rimraf altogether, but I'd like to keep it for another time.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -9,38 +9,20 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [16.x]
+        node-version: [16]
```
Comment (janhalama) [inline .github/workflows/testing.yml:12]: suggestion: Node.js 16 is going to be deprecated on 2023-09-11 I suggest to switch tests to run on LTS version which is 18.

**Review by janhalama — APPROVED** [review]
I have one minor suggestion otherwise looks good 👍

File: .github/workflows/testing.yml (hunk 1 above)
Comment (jnv) [reply .github/workflows/testing.yml:12]: I think it's risky to not test on the currently popular version. In fact, I've intentionally targeted tsc to Node 12,so maybe expand the build matrix to include 12, 14, 16, and 18?

Comment (janhalama) [reply .github/workflows/testing.yml:12]: I agree lets expand the matrix as you suggest.

**Review by janhalama — APPROVED** [review]
LGTM

**Pull request merged**